apps/rush-lib/src/logic/TaskSelector.ts

```diff
@@ -42,71 +39,67 @@ export class TaskSelector {
-      );
-    }
   }
 
   public registerTasks(): TaskCollection {
-    if (this._options.toFlags.length > 0) {
-      this._registerToFlags(this._options.toFlags);
+    if (this._options.toProjects.length > 0) {
+      this._registerToProjects(this._options.toProjects);
     }
-    if (this._options.fromFlags.length > 0) {
-      this._registerFromFlags(this._options.fromFlags);
+    if (this._options.fromProjects.length > 0) {
+      this._registerFromProjects(this._options.fromProjects);
     }
-    if (this._options.toFlags.length === 0 && this._options.fromFlags.length === 0) {
+    if (this._options.toProjects.length === 0 && this._options.fromProjects.length === 0) {
       this._registerAll();
     }
 
     return this._taskCollection;
   }
 
-  private _registerToFlags(toFlags: ReadonlyArray<string>): void {
-    const dependencies: Set<string> = new Set<string>();
+  private _registerToProjects(toProjects: ReadonlyArray<RushConfigurationProject>): void {
+    const dependencies: Map<string, RushConfigurationProject> = new Map<string, RushConfigurationProject>();
 
-    for (const toFlag of toFlags) {
-      this._collectAllDependencies(toFlag, dependencies);
+    for (const toProject of toProjects) {
+      this._collectAllDependencies(toProject, dependencies);
     }
 
     // Register any dependencies it may have
-    for (const dependency of dependencies) {
-      this._registerTask(this._options.rushConfiguration.getProjectByName(dependency));
+    for (const [, dependencyProject] of dependencies) {
+      this._registerTask(dependencyProject);
     }
 
     if (!this._options.ignoreDependencyOrder) {
       // Add ordering relationships for each dependency
-      for (const dependency of dependencies) {
-        this._taskCollection.addDependencies(dependency, this._rushLinkJson.localLinks[dependency] || []);
+      for (const [, dependencyProject] of dependencies) {
+        this._taskCollection.addDependencies(
+          ProjectTask.getTaskName(dependencyProject),
+          dependencyProject.localDependencyProjects.map((x) => ProjectTask.getTaskName(x))
+        );
       }
     }
   }
 
-  private _registerFromFlags(fromFlags: ReadonlyArray<string>): void {
+  private _registerFromProjects(fromProjects: ReadonlyArray<RushConfigurationProject>): void {
     this._buildDependentGraph();
-    const dependents: Set<string> = new Set<string>();
+    const dependents: Map<string, RushConfigurationProject> = new Map<string, RushConfigurationProject>();
 
-    for (const fromFlag of fromFlags) {
-      this._collectAllDependents(fromFlag, dependents);
+    for (const fromProject of fromProjects) {
+      this._collectAllDependents(fromProject, dependents);
     }
 
     // Register all downstream dependents
-    for (const dependent of dependents) {
-      this._registerTask(this._options.rushConfiguration.getProjectByName(dependent));
+    for (const [, dependentProject] of dependents) {
```

Comment:
`[, dependentProject]` is pretty weird notation -- can you just use `.keys()` or `.values()`?